cmd/system-probe/modules/security_runtime.go

```diff
@@ -9,15 +9,28 @@ import (
 	"github.com/DataDog/datadog-agent/cmd/system-probe/api"
 	"github.com/DataDog/datadog-agent/pkg/ebpf"
 	"github.com/DataDog/datadog-agent/pkg/process/config"
+	sconfig "github.com/DataDog/datadog-agent/pkg/security/config"
 	secmodule "github.com/DataDog/datadog-agent/pkg/security/module"
+
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // SecurityRuntime - Security runtime Factory
 var SecurityRuntime = api.Factory{
 	Name: "security_runtime",
-	Fn: func(cfg *config.AgentConfig) (api.Module, error) {
-		module, err := secmodule.NewModule(cfg)
+	Fn: func(agentConfig *config.AgentConfig) (api.Module, error) {
+		config, err := sconfig.NewConfig(agentConfig)
+		if err != nil {
+			log.Errorf("invalid security runtime module configuration: %s", err)
```

Comment:
it feels like returned errors from a factory.Fn should get logged in a uniform way instead of relying on each module to handle logging errors

Reply:
Right. Fixed